integrations/dynatrace/main.py

```diff
@@ -0,0 +1,70 @@
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import DynatraceClient
+
+
+class ObjectKind(StrEnum):
+    PROBLEM = "problem"
+    SLO = "slo"
+    ENTITY = "entity"
+
+
+def initialize_client() -> DynatraceClient:
+    return DynatraceClient(
+        ocean.integration_config["dynatrace_host_url"],
+        ocean.integration_config["dynatrace_api_key"],
+    )
+
+
+@ocean.on_resync(ObjectKind.PROBLEM)
+async def on_resync_problems(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dynatrace_client = initialize_client()
+
+    async for problems in dynatrace_client.get_problems():
+        logger.info(f"Received batch with {len(problems)} problems")
+        yield problems
+
+
+@ocean.on_resync(ObjectKind.SLO)
+async def on_resync_slos(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dynatrace_client = initialize_client()
+
+    async for slos in dynatrace_client.get_slos():
+        logger.info(f"Received batch with {len(slos)} SLOs")
```

Comment:
you are repeating this log line in each resource in the `_get_paginated_resources`  with including the `resource_key` in the log to indicate to what resource it is related to.

Reply:
I'm sorry, I don't understand what you mean

Reply:
In the     ```
async def _get_paginated_resources(
        self, url: str, resource_key: str, params: dict[str, Any] = {}
    ) -> AsyncGenerator[list[dict[str, Any]], None]:
```
       
        
        You are already logging the resources key
         `logger.info(f"Fetching {resource_key} from {url} with params {params}")`
         
         So I believe the comment is to remove the log line in the 
        ` logger.info(f"Received batch with {len(slos)} SLOs")`